src/containers/PublicKey/index.tsx

```diff
@@ -0,0 +1,81 @@
+import { utils } from "ethers";
+import { useState } from "react";
+import { useWeb3 } from "../../web3";
+import Button from "../../components/layout/Button";
+import icon from "../../assets/images/copy.svg";
+import { toast } from "react-toastify";
+import { Heading } from "../../assets/styles/headings.enum";
+
+const PublicKey = () => {
+  const [publicKey, setPublicKey] = useState("");
+  const { account, signerOrProvider } = useWeb3();
+
+  const getPublicKey = async () => {
+    if (!signerOrProvider) return;
+    try {
+      // todo needed to add this line to prevent type error
+      // todo this will need to be updated to handle this situation better
+      const signer: any = signerOrProvider;
+      const msg = "Hello from the Sarcophagus! Sign this message to retrieve your account's public key";
+      const msgHash = utils.hashMessage(msg);
+      const msgHashBytes = utils.arrayify(msgHash);
+      const signature = await signer?.signMessage(
```

Comment:
I think this null check on `signer` should be removed, and if the signer doesn't exist an error will be thrown.